app/src/main/java/org/glucosio/android/activity/AddWeightActivity.java

```diff
@@ -62,14 +66,18 @@ protected void onCreate(Bundle savedInstanceState) {
 
         if (!"kilograms".equals(presenter.getWeightUnitMeasuerement())) {
             unitTextView.setText("lbs");
+            needUnitConversion = true;
         }
 
         // If an id is passed, open the activity in edit mode
         FormatDateTime formatDateTime = new FormatDateTime(getApplicationContext());
         if (this.isEditing()) {
             setTitle(R.string.title_activity_add_weight_edit);
             WeightReading readingToEdit = presenter.getWeightReadingById(this.getEditId());
-            readingTextView.setText(readingToEdit.getReading() + "");
+            double weightVal = readingToEdit.getReading();
+            if (needUnitConversion)
+                weightVal = GlucosioConverter.kgToLb(weightVal);
```

Comment:
Was it somehow forgotten?

Reply:
I don't understand this comment

Reply:
Sorry for that. The question was - we never had this functionality so we never had a proper working weight for imperial countries, correct?

Reply:
Correct, as far as I can tell it never worked for imperial countries

Reply:
Clear! 👍